mwdb/web/src/components/Profile/ProfileViews.js

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { Link, Route, Switch, useHistory, useParams } from "react-router-dom";
+
+import api from "@mwdb-web/commons/api";
+import { AuthContext } from "@mwdb-web/commons/auth";
+import { View, getErrorMessage } from "@mwdb-web/commons/ui";
+
+import ProfileDetails from "./Views/ProfileDetails";
+import ProfileAPIKeys from "./Views/ProfileAPIKeys";
+import ProfileCapabilities from "./Views/ProfileCapabilities";
+import ProfileResetPassword from "./Views/ProfileResetPassword";
+import ProfileGroup from "./Views/ProfileGroup";
+
+export default function ProfileViews() {
+    const auth = useContext(AuthContext);
+    const history = useHistory();
+    const user = useParams().user || auth.user.login;
+    const [profile, setProfile] = useState();
+
+    async function updateProfile() {
+        try {
+            const response = await api.getUserProfile(user);
+            setProfile(response.data);
+        } catch (error) {
+            history.push({
+                pathname: "/profile",
+                state: { error: getErrorMessage(error) },
+            });
+        }
+    }
+
+    function GroupBreadcrumb() {
+        const { group } = useParams();
+        return `Group '${group}' details`;
+    }
+
+    function UserBreadcrumb() {
+        const { user } = useParams();
+        return `User '${user}' details`;
+    }
+
+    const getProfile = useCallback(updateProfile, [user]);
+
+    useEffect(() => {
+        getProfile();
+    }, [getProfile]);
+
+    if (!profile) return [];
```

Comment:
This section doesn't quite work.

Possibly there is a problem with getting parameters by:
`    const user = useParams().user || auth.user.login;`

On every profile view we have constantly auth.login profile.
It's not change when you try to get another user's details information.

Something is wrong with getting user from params or with lifecycle in this section.

Reply:
Fixed :+1: https://github.com/CERT-Polska/mwdb-core/pull/337/commits/e7be9aee85060996307a45228d59c310386c8847